sdk/tables/azure-data-tables/tests/test_custom_entity.py

```diff
@@ -0,0 +1,422 @@
+        if "Z" in bday:
+            # Has milliseconds, we'll handle this separately
+            # Python library only handles 6 decimal places of precision
+            ms = bday.split(".")[-1]
+            ms = "." + ms[:-1]
+            ms = float(ms)
+            date = bday.split(".")[0]
+            date = datetime.strptime(date, "%Y-%m-%dT%H:%M:%S")
+            return date, ms
+        else:
+            return datetime.strptime(date, "%Y-%m-%dT%H:%M:%S"), 0.0
+
+    def keys(self):
+        return self.properties.keys()
+
+
+def modify_returned_class(entity):
+    if "birthday" in entity.keys():
+        bday = entity["birthday"]
+        if "Z" in bday:
+            ms = bday.split(":")[-1]
+            ms = "." + ms
+            ms = float(ms)
+            date = bday.replace(str(ms), "")
+            date = datetime.strptime(date, "%Y-%m-%dT%H:%M")
+            entity["birthday"] = date
+            entity["birthday_ms"] = ms
+    if "Birthday" in entity.keys():
+        bday = entity["birthday"]
+        if "Z" in bday:
+            ms = bday.split(":")[-1]
+            ms = "." + ms
+            ms = float(ms)
+            date = bday.replace(str(ms), "")
+            date = datetime.strptime(date, "%Y-%m-%dT%H:%M")
+            entity["Birthday"] = date
+            entity["Birthday_ms"] = ms
+
+    return entity
+
+
+class StorageTableEntityTest(TableTestCase):
+    def _set_up(
+        self,
+        tables_storage_account_name,
+        tables_primary_storage_account_key,
+        url="table",
+    ):
+        table_name = self.get_resource_name("uttable")
+        ts = TableServiceClient(
+            self.account_url(tables_storage_account_name, url),
+            credential=tables_primary_storage_account_key,
+            table_name=table_name,
+        )
+        table = ts.get_table_client(table_name)
+        if self.is_live:
+            try:
+                ts.create_table(table_name)
+            except ResourceExistsError:
+                pass
+        return table
+
+    def _tear_down(
+        self,
+        tables_storage_account_name,
+        tables_primary_storage_account_key,
+        url="table",
+    ):
+        table_name = self.get_resource_name("uttable")
+        ts = TableServiceClient(
+            self.account_url(tables_storage_account_name, url),
+            credential=tables_primary_storage_account_key,
+            table_name=table_name,
+        )
+        if self.is_live:
+            try:
+                ts.delete_table(table_name)
+            except:
+                pass
+
+    @TablesPreparer()
+    def test_custom_entity(
+        self, tables_storage_account_name, tables_primary_storage_account_key
+    ):
+        table_client = self._set_up(
+            tables_storage_account_name, tables_primary_storage_account_key
+        )
+
+        entity = {
+            u"PartitionKey": u"pk",
+            u"RowKey": u"rk",
+            u"Birthday": u"2020-01-01T12:59:59.0123456Z",
```

Comment:
Could we also try (maybe in a separate test) doing this with an edmtype?
IIRC - it would be something like:
`u"Birthday": EdmType(EdmTypes.Datetime, u"2020-01-01T12:59:59.0123456Z")`

Reply:
This will raise a failure because the `EntityProperty` will force it to be a `datetime.datetime` object and this will fail on the seven decimal places.

Reply:
Let's have EntityProperty default to passing through strings rather than attempt to manipulate them into datetime.